### PR TITLE
Changed Loader Closure definition

### PR DESCRIPTION
**Pull request by norberttech**
<h2>Change Log</h2>
<div id="change-log">
  <h4>Added</h4>
  <ul id="added">
    <li>batchSize argument to DataFrame::collect method</li>
  </ul> 
  <h4>Fixed</h4>  
  <ul id="fixed">
    
  </ul>
  <h4>Changed</h4>
  <ul id="changed">
    <li>Closure was moved to Loader namespace as it applies to Loaders, not Pipelines</li>
  </ul>  
  <h4>Removed</h4>
  <ul id="removed">
    <li>Rows from Closure::closure method</li>
    <li>BufferLoader</li>
  </ul>
  <h4>Deprecated</h4>
  <ul id="deprecated">
    
  </ul>  
  <h4>Security</h4> 
  <ul id="security">
    
  </ul>     
</div>
<hr/>

Ref: #715 

<h2>Description</h2>

BufferLoader was implemented at very early stage of ETL, it's goal was to reduce number of IO operations when writing data. From now the same behavior can be achieved by setting a batchSize on `DataFrame::collect()` method. Before collect was collecting everything into memory, and it still does that when batchSize is not specified. 
When batchSize is defined, CollectingPipeline will keep merging rows until they reach batchSize and then it will yield them. 

Why is this important? 

Thanks to the collect method, users have control over how many Rows at once are passed to the Loader, which can help in achieving better performance or displaying more readable output. 

We can also remove batchSize from the Loaders directly, as they should not be responsible for dividing received Rows into chunks, but it's a thing for the next PR.

### CODE REVIEW

File: src/adapter/etl-adapter-avro/src/Flow/ETL/Adapter/Avro/FlixTech/AvroLoader.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ public function __unserialize(array $data) : void
         $this->writer = null;
     }
 
-    public function closure(Rows $rows, FlowContext $context) : void
+    public function closure(FlowContext $context) : void
```
Comment (stloyd) [inline src/adapter/etl-adapter-avro/src/Flow/ETL/Adapter/Avro/FlixTech/AvroLoader.php:52]: As this is public I think it also requires note in upgrade file.

File: src/core/etl/src/Flow/ETL/DSL/To.php [hunk 2]
```diff
@@ -18,14 +18,6 @@
  */
 class To
 {
-    /**
-     * @param int<1, max> $bufferSize
-     */
-    final public static function buffer(Loader $overflowLoader, int $bufferSize) : Loader
```
Comment (stloyd) [inline src/core/etl/src/Flow/ETL/DSL/To.php:24]: This most likely also requires note in upgrade file.

**Review by stloyd — COMMENTED** [review]
I like the changes, just small notes that seem to be missing from the upgrade file

File: src/core/etl/src/Flow/ETL/DSL/To.php (hunk 2 above)
Comment (norberttech) [reply src/core/etl/src/Flow/ETL/DSL/To.php:24]: it's already there

**Comment by norberttech** [comment]
Two more things can be implemented after this is merged: 

#717 
#718

File: src/core/etl/src/Flow/ETL/DSL/To.php (hunk 2 above)
Comment (stloyd) [reply src/core/etl/src/Flow/ETL/DSL/To.php:24]: It says about `BufferLoader`, but not about the `To::buffer()` method, which is related but still different place.

Comment (norberttech) [reply src/core/etl/src/Flow/ETL/DSL/To.php:24]: it's the same thing, To::buffer is only a DSL function to simplify initialization of BufferLoader but feel free to add it into the upgrade notes

**Pull request merged**